subchannel_test.go

```diff
@@ -252,3 +253,29 @@ func TestGetSubchannelOptionsOnNew(t *testing.T) {
 	require.NoError(t, err, "Should get peer")
 	assert.Equal(t, want, peer, "Unexpected peer")
 }
+
+func TestHandlerWithoutSubChannel(t *testing.T) {
+	ch, err := NewChannel("svc", &ChannelOptions{
```

Comment:
Few suggests for this test:
 - use `testutils` to reduce the boilerplate in this test. You might want to add the new handler option to the `ChannelOptions` as well, so you can pass it in.
 - test the root channel name + a few subchannels
 - if you register an echo handler, you can use `testutils.AssertEcho`

```go
handler := HandlerFunc(...)
opts := testutils.NewOpts().[...]
testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
  client := ts.NewClient(nil)
  for _, svc := range []string{ts.ServiceName(), "misc", "other"} {
    testutils.AssertEcho(t, client, ts.HostPort(), svc)
  }
})
```